src/io/github/humbleui/canvas.clj

```diff
@@ -1,8 +1,14 @@
 (ns io.github.humbleui.canvas
   (:import
-    [io.github.humbleui.types IRect Rect RRect]
+    [io.github.humbleui.types Point IRect Rect RRect]
     [io.github.humbleui.skija BlendMode Canvas Font Paint]))
 
+(defn draw-polygon
+  ([^Canvas canvas floats-or-points ^Paint paint]
+   (if (= (type floats-or-points) "[F")
+     (.drawPolygon canvas #^floats floats-or-points paint)
+     (.drawPolygon canvas ^"[Lio.github.humbleui.types.Point;" floats-or-points paint))))
```

Comment:
I'm not super happy with this type hinting approach, so if you have a better suggestion I'm happy to hear it.

This was the only way I could correctly hint the different methods without taking a reflection hit.

There doesn't seem to be an elegant way to Array hint?

Reply:
Maybe two different methods? One that accepts floats and one that accepts points?

Reply:
Hmm, not convinced that's much better? I'll leave it this way for a bit, see if it sparks a better idea / approach. Worst case we can look at doing that.

Reply:
Oh @tonsky, I've noticed that there's no open / closed mode for polygon drawing, it seems to assume that it's open.
Do you want to support it? Or not bother for the moment?

Reply:
I suspect you’re supposed to do that with Path. Path has closePath method. I’d like to stay as close to Skia API as possible

Reply:
Sorry, do you mean don't worry about close for polygon drawing? Or use `Path` to provide this?

Reply:
don't worry about close for polygon drawing